<?xml version="1.0"?>
<launch>

  <!-- DBW system -->
  <arg name="sys" default="false"/>
  <arg name="dataspeed_can" default="true" />
  <arg name="socketcan" default="false" />
  <arg name="socketcan_dev" default="can0" />
  <arg name="buttons" default="true"/>
  <group if="$(var sys)">
    <include file="$(find-pkg-share ds_dbw_can)/launch/dbw.launch.xml" >
      <arg name="dataspeed_can" value="$(var dataspeed_can)" />
      <arg name="socketcan" value="$(var socketcan)" />
      <arg name="socketcan_dev" value="$(var socketcan_dev)" />
      <arg name="buttons" value="$(var buttons)" />
    </include>
  </group>

  <!-- Joystick demo -->
  <arg name="enable" default="true"/> <!-- Enable buttons -->
  <arg name="ignore" default="false"/> <!-- Ignore driver overrides -->
  <arg name="steer" default="true"/> <!-- Send steering commands -->
  <arg name="brake" default="true"/> <!-- Send brake commands -->
  <arg name="thrtl" default="true"/> <!-- Send throttle commands -->
  <arg name="shift" default="true"/> <!-- Send shift commands -->
  <arg name="misc" default="true"/> <!-- Send misc commands -->
  <arg name="steer_cmd_type" default=""/> <!-- Select steering command type -->
  <arg name="brake_cmd_type" default=""/> <!-- Select brake command type -->
  <arg name="thrtl_cmd_type" default=""/> <!-- Select throttle command type -->
  <arg name="steer_max" default=""/> <!-- Maximum steering command value (changes with cmd_type) -->
  <arg name="brake_min" default=""/> <!-- Minimum brake command value (changes with cmd_type) -->
  <arg name="brake_max" default=""/> <!-- Maximum brake command value (changes with cmd_type) -->
  <arg name="thrtl_min" default=""/> <!-- Minimum throttle command value (changes with cmd_type) -->
  <arg name="thrtl_max" default=""/> <!-- Maximum throttle command value (changes with cmd_type) -->
  <arg name="steer_rate" default="0.0"/> <!-- Steering rate limit -->
  <arg name="steer_accel" default="0.0"/> <!-- Steering acceleration limit -->
  <arg name="brake_inc" default="0.0"/> <!-- Brake rate increase limit (changes with cmd_type) -->
  <arg name="brake_dec" default="0.0"/> <!-- Brake rate decrease limit (changes with cmd_type) -->
  <arg name="thrtl_inc" default="0.0"/> <!-- Throttle rate increase limit (changes with cmd_type) -->
  <arg name="thrtl_dec" default="0.0"/> <!-- Throttle rate decrease limit (changes with cmd_type) -->
  <node namespace="vehicle" pkg="ds_dbw_joystick_demo" exec="joy_demo" output="screen" >
    <param name="enable" value="$(var enable)" />
    <param name="ignore" value="$(var ignore)" />
    <param name="steer" value="$(var steer)" />
    <param name="brake" value="$(var brake)" />
    <param name="thrtl" value="$(var thrtl)" />
    <param name="shift" value="$(var shift)" />
    <param name="misc" value="$(var misc)" />
    <param name="steer_cmd_type" value="$(var steer_cmd_type)"/>
    <param name="brake_cmd_type" value="$(var brake_cmd_type)"/>
    <param name="thrtl_cmd_type" value="$(var thrtl_cmd_type)"/>
    <param name="steer_max" value="$(var steer_max)"/>
    <param name="brake_min" value="$(var brake_min)"/>
    <param name="brake_max" value="$(var brake_max)"/>
    <param name="thrtl_min" value="$(var thrtl_min)"/>
    <param name="thrtl_max" value="$(var thrtl_max)"/>
    <param name="steer_rate" value="$(var steer_rate)"/>
    <param name="steer_accel" value="$(var steer_accel)"/>
    <param name="brake_inc" value="$(var brake_inc)"/>
    <param name="brake_dec" value="$(var brake_dec)"/>
    <param name="thrtl_inc" value="$(var thrtl_inc)"/>
    <param name="thrtl_dec" value="$(var thrtl_dec)"/>
  </node>

  <!-- Joystick driver -->
  <arg name="device_id" default="0"/>
  <node pkg="joy" exec="joy_node" name="joy_node" output="screen" >
    <param name="device_id" value="$(var device_id)" />
    <param name="deadzone" value="0.0" />
    <param name="autorepeat_rate" value="50.0" />
  </node>

</launch>
